### PR TITLE
#7Task. Ivan Menshov

### PR DESCRIPTION
**Pull request by IvanMenshov**
https://travis-ci.com/github/IvanMenshov/js-assignments/builds/175558802

### CODE REVIEW

File: task/07-yield-tasks.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+debugger;
```
Comment (AlexMokin) [inline task/07-yield-tasks.js:1]: remove debugger statment

File: task/07-yield-tasks.js [hunk 2]
```diff
@@ -126,7 +171,33 @@ function* breadthTraversalTree(root) {
  *   [ 1, 3, 5, ... ], [ -1 ] => [ -1, 1, 3, 5, ...]
  */
 function* mergeSortedSequences(source1, source2) {
-    throw new Error('Not implemented');
+    var first = source1();
+    var second = source2();
+    var a = first.next();
+    var b = second.next();
+
+    while(true){
```
Comment (AlexMokin) [inline task/07-yield-tasks.js:179]: add exit condition for a while loop for the case when sources are not infinite

File: task/07-yield-tasks.js (hunk 1 above)
Comment (AlexMokin) [inline task/07-yield-tasks.js:1]: remove debugger statment

File: task/07-yield-tasks.js [hunk 3]
```diff
@@ -126,7 +171,33 @@ function* breadthTraversalTree(root) {
...
+
+        if(a.done){
+            yield b.value;
+            b = second.next();
```
Comment (AlexMokin) [inline task/07-yield-tasks.js:183]: what if both a.done and b.done are true?

**Comment by IlyaMokin** [comment]
Thanks! Closing as accepted!

**Pull request closed without merging**